bc/channel/utils/connectors/bluesky_api/client.py

```diff
@@ -313,9 +357,17 @@ def post_status(self, text: str, media: list[Thumbnail]) -> dict[str, str]:
                 "images": media,
             }
         elif message_object["facets"]:
-            card = self.fetch_embed_url_card(
-                message_object["facets"][-1]["features"][0]["uri"]
-            )
+            link: str
+            card: SocialCard | None
```

Comment:
Let's assign a default value to these new variables. This will help us avoid the `UnboundLocalError` which can occur if the variable is referenced before it's explicitly assigned a value within the code's scope. Providing a default value ensures the variable is always initialized and has a defined state.

```suggestion
            link: str | None = None
            card: SocialCard | None = None
```

Reply:
I would assume that checking if the value isn't None would narrow the type down but Mypy seems to be arguing otherwise, is there something I'm missing on?

![image](https://github.com/freelawproject/bigcases2/assets/148872143/7a48786e-f4fd-482b-bb6f-950fcae89885)

Reply:
Ah, dropping the `!= None` works.